internal/pkg/agent/install/wait_service_windows.go

```diff
@@ -58,3 +60,41 @@ func isStopped(timeout time.Duration, interval time.Duration, service string) er
 		}
 	}
 }
+
+// EnsureServiceRemoved opens the Windows service manager and checks if the
+// service is removed. It will repeat this check every 'interval'
+// until the 'timeout' is reached.  It returns nil if the service
+// is removed within the timeout period.  An error is returned if
+// the service is not removed before the timeout or if there are
+// errors communicating with the service manager.
+func EnsureServiceRemoved(timeout time.Duration, interval time.Duration, service string) error {
+	m, err := mgr.Connect()
+	if err != nil {
+		return fmt.Errorf("failed to connect to service manager: %w", err)
+	}
+
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			s, err := m.OpenService(service)
+			switch {
+			case err == nil:
+				// The service is still installed continue waiting
+				_ = s.Close()
+			case errors.Is(err, windows.ERROR_SERVICE_DOES_NOT_EXIST):
+				// The service is no longer installed
+				return nil
+			default:
+				// An unknown error occurred trying to open the service
+				return fmt.Errorf("error opening service (%s): %w", service, err)
+			}
+		case <-timer.C:
+			return fmt.Errorf("timed out after %s waiting for service (%s) to uninstall", timeout, service)
```

Comment:
This path doesn't call `s.Close()`

Reply:
That will leak a service handle:

```go
// Close relinquish access to the service s.
func (s *Service) Close() error {
	return windows.CloseServiceHandle(s.Handle)
}
```

Reply:
hmm 🤔 I don't see this problem, I do open the service and close it when there is no error; according to [this](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-openservicew) (underlying windows call of OpenService)  on error the handle is NULL but I guess you say `better safe than sorry always close it`?!

However your comment made me look again, and I am not disconnecting from the `service control manager` when I go out of the scope of this function, so thanks 🙂

Reply:
Yeah this looks to be calling `s.Close()` in the case that there is no error, which looks correct. Glad it help find them missing call to `m.Disconnect()`. I missed that as well in my review. I always forget that it also has to be called.